Simbody/src/GeneralForceSubsystem.cpp

```diff
@@ -123,30 +369,43 @@ class GeneralForceSubsystemRep : public ForceSubsystem::Guts {
 
         // Some forces are disabled by default; initialize the enabled flags
         // accordingly. Also, see if we're going to need to do any caching
-        // on behalf of any forces that don't depend on velocities. 
+        // on behalf of any forces that don't depend on velocities.
         Array_<bool> forceEnabled(getNumForces());
         bool someForceElementNeedsCaching = false;
         for (int i = 0; i < (int)forces.size(); ++i) {
             forceEnabled[i] = !(forces[i]->isDisabledByDefault());
             if (!someForceElementNeedsCaching)
-                someForceElementNeedsCaching = 
+                someForceElementNeedsCaching =
                     forces[i]->getImpl().dependsOnlyOnPositions();
         }
 
-        forceEnabledIndex = allocateDiscreteVariable(s, Stage::Instance, 
+        forceEnabledIndex = allocateDiscreteVariable(s, Stage::Instance,
             new Value<Array_<bool> >(forceEnabled));
 
-        // Note that we'll allocate these even if all the needs-caching 
+        //Set the force's parallelization flags accordingly, do not readd the forces to the vector
+        //since a force is not able to change its parallelism flag mid-computation
+        //TODO: Previous optimization only computed parallel once - but error in testInitState - why?
+        // if(!computedParallel){
+            Array_<bool> parallelEnabled(getNumForces());
+            for (int i = 0; i < (int)forces.size(); ++i){
+                parallelEnabled[i] = forces[i]->getImpl().isParallelByDefault();
+            }
+            parallelEnabledIndex = allocateDiscreteVariable(s, Stage::Instance,
```

Comment:
I don't understand why `parallelEnabled` is a discrete state variable (like the "enabled" flag). That implies it is something a user can change in the state (like enabling and disabling a force). Is there a reason that needs to be changed on the fly?